product_packaging_container_deposit/models/product_product.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2023 Camptocamp (<https://www.camptocamp.com>).
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html)
+
+from odoo import models
+from odoo.tools import groupby
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    def get_product_container_deposit_quantities(self, qty, forced_packaging=False):
+        """
+        Returns the quantity of deposit per packaging level for a given product quantity
+        {
+            "PL1": (CP1, QTY),
+            "PLn": (CPn, QTYn)
+        }
+        """
+
+        def get_sort_key(packaging):
```

Comment:
there's no need to sort before grouping when using the `groupby` from odoo.tools. 
It's needed when using `itertools.groupby`

Reply:
without sort it will break `test_product_container_deposit_quantities_per_packaging_level`

Reply:
if you need it for tests only I'd say it's not needed. You can sort results in the test.
If it's needed for functional purposes.... fine, keep it.